virtual_rainforest/models/plants/plants_model.py

```diff
@@ -108,18 +122,29 @@ def from_config(
             config: A validated :class:`~virtual_rainforest.core.config.Config`
                 instance.
             update_interval: Frequency with which all models are updated
+
         """
 
-        # Check if any constants have been supplied
+        # Get required init arguments from config
+        soil_layers = config["core"]["layers"]["soil_layers"]
+        canopy_layers = config["core"]["layers"]["canopy_layers"]
+
         # Load in the relevant constants
         constants = load_constants(config, "plants", "PlantsConsts")
 
         # Generate the flora
         flora = Flora.from_config(config=config)
 
-        # Try and create the instance - safeguard against exceptions
+        # Try and create the instance - safeguard against exceptions from __init__
```

Comment:
Out of interest, does the code that creates model instances not already check for errors raised during this process? If not, might that be a better place to catch errors than in model classes' factory methods?

Reply:
Hmm - I had a reason. It felt clever at the time, but I cannot remember it for the life of me. Might be the heat. 

I think it was along the lines of wanting users to understand that the process of turning their configuration into a model instance had borked, and hopefully giving them some context on where to fix things? @vgro @jacobcook1995 @TaranRallings ?

Reply:
That's handled for use from `vr_run`, but people might be creating these models from within a Python session. I agree that is going to be much less common, but catching errors here isn't a bad thing. @alexdewar - is that what you meant? That _upstream_ code creating a suite of model instances would catch them? I thought you meant _downstream_ code in the `__init__` or somewhere in `BaseModel`?

Reply:
That was basically my thinking - make sure _something_ gets written to the log if it falls over.

Reply:
Will leave as is for the moment.